src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -15,6 +16,7 @@ internal class DefaultStorageAccountProvider : IStorageAccountProvider
         private readonly IStorageCredentialsValidator _storageCredentialsValidator;
         private readonly IStorageAccountParser _storageAccountParser;
         private readonly IServiceProvider _services;
+        private readonly ConcurrentDictionary<string, Task<IStorageAccount>> _accounts = new ConcurrentDictionary<string, Task<IStorageAccount>>();
```

Comment:
All right. While it is a little strange to have a cache of completed tasks, it should work ok. I would have just done something a bit more traditional, etc.:

```csharp
public async Task<IStorageAccount> TryGetAccountAsync(string connectionStringName, CancellationToken cancellationToken)
{
    IStorageAccount account;
    if (!_accounts.TryGetValue(connectionStringName, out account))
    {   
        account = await CreateAndValidateAccountAsync(connectionStringName, cancellationToken);
        _accounts.AddOrUpdate(connectionStringName, (cs) => account, (cs, a) => account);
    }
    return account;
}
```

From time to time we might end up validating an account twice with that code, but the value is cached quickly and this solves the issue.